### PR TITLE
Update list.sh for busybox compatibility

### PR DESCRIPTION
**Pull request by diginc**
I believe this has feature parity with `sed /foo/ Id` but also supports busybox, and my alpine docker ;)

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

_8_

---
uses grep to find the line numbers matching the domain (case insensitive) and feeds them into sed for deletion.

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

File: advanced/Scripts/list.sh [hunk 1]
```diff
@@ -157,8 +157,8 @@ RemoveDomain() {
         if [[ "${bool}" == true ]]; then
             # Remove it from the other one
             echo "::: Removing $1 from $list..."
-            # /I flag: search case-insensitive
-            sed -i "/${domain}/Id" "${list}"
+            # Busybox sed compatible case-insensitive domain removal
+            sed -i "$(grep -in "^${domain}$" ${list} | awk -F':' '{print $1}' | tr '\n' ',' | sed 's/,$/\n/')d" ${list}
```
Comment (dschaper) [inline advanced/Scripts/list.sh]: Does `^${domain}$` work even though the lists start with more information before the domains are listed?

Comment (diginc) [inline advanced/Scripts/list.sh]: I thought the gravity.list had the IPs on the front, but whitelist/blacklists just had the domain them selves, one per line.

Comment (dschaper) [inline advanced/Scripts/list.sh]: It does, I was testing against the wrong lists, I was using the source lists, and got switched up.

Comment (dschaper) [inline advanced/Scripts/list.sh]: Edited comment to strikeout and clarify.

**Pull request merged**